pyads/ads.py

```diff
@@ -418,6 +447,203 @@ def set_timeout(ms):
+
+    :return: c_ubyte_Array: data size required to read/write a structure of multiple types
+    """
+    num_of_bytes = 0
+    for item in structure_def:
+        try:
+            var, plc_datatype, size = item
+            str_len = None
+        except ValueError:
+            var, plc_datatype, size, str_len = item
+
+        if plc_datatype == PLCTYPE_STRING:
+            if str_len is not None:
+                num_of_bytes += (str_len + 1) * size
+            else:
+                num_of_bytes += (PLC_DEFAULT_STRING_SIZE + 1) * size
+        elif plc_datatype in PLC_8_BYTE_TYPES:
+            num_of_bytes += 8 * size
+        elif plc_datatype in PLC_4_BYTE_TYPES:
+            num_of_bytes += 4 * size
+        elif plc_datatype in PLC_2_BYTE_TYPES:
+            num_of_bytes += 2 * size
+        elif plc_datatype in PLC_BYTE_TYPES:
+            num_of_bytes += 1 * size
+        else:
+            raise RuntimeError("Datatype not found")
+
+    return c_ubyte * num_of_bytes
+
+
+def dict_from_bytes(byte_list, structure_def, array_size=1):
+    """Return an ordered dict of PLC values from a list of BYTE values read from PLC.
+
+    :param byte_list: list of byte values for an entire structure
+    :param tuple structure_def: special tuple defining the structure and
+            types contained within it according o PLCTYPE constants
+
+            Expected input example:
+
+            structure_def = (
+                ('rVar', pyads.PLCTYPE_LREAL, 1),
+                ('sVar', pyads.PLCTYPE_STRING, 2, 35),
+                ('rVar1', pyads.PLCTYPE_REAL, 1),
+                ('iVar', pyads.PLCTYPE_DINT, 1),
+                ('iVar1', pyads.PLCTYPE_INT, 3),
+                ('ivar2', pyads.PLCTYPE_UDINT, 1),
+                ('iVar3', pyads.PLCTYPE_UINT, 1),
+                ('iVar4', pyads.PLCTYPE_BYTE, 1),
+                ('iVar5', pyads.PLCTYPE_SINT, 1),
+                ('iVar6', pyads.PLCTYPE_USINT, 1),
+                ('bVar', pyads.PLCTYPE_BOOL, 4),
+                ('iVar7', pyads.PLCTYPE_WORD, 1),
+                ('iVar8', pyads.PLCTYPE_DWORD, 1),
+            )
+            i.e ('Variable Name', variable type, arr size (1 if not array),
+                 length of string (if defined in PLC))
+
+    :param array_size: size of array if reading array of structure, defaults to 1
+    :type array_size: int, optional
+
+    :return: ordered dictionary of values for each variable type in order of structure
+    """
+    values_list = []
+    index = 0
+    for structure in range(0, array_size):
+        values = OrderedDict()
+        for item in structure_def:
+            try:
+                var, plc_datatype, size = item
+                str_len = None
+            except ValueError:
+                var, plc_datatype, size, str_len = item
+
+            if size == 1:  # add straight to dictionary if not array
+                if plc_datatype == PLCTYPE_STRING:
+                    if str_len is not None:

```

Comment:
Just use if str_len is None and omit the else so there is no double negative condition